pkg/components/velero/component.go

```diff
@@ -189,15 +193,15 @@ func (c *component) getSupportedProviders() []string {
 func (c *component) getProvider() (provider, error) {
 	providers := []provider{}
 
-	if c.Azure != nil {
+	if c.Provider == "azure" {
```

Comment:
Actually, append logic doesn't make sense anymore if we can only have one provider configured at all times. I'd just replace entire function body with either switch/case or map lookup.